src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const splittedDate = date.split(separator);
+  const lengthSplitedDate = splittedDate.length;
+  const storeDateObj = {};
+
+  for (let i = 0; i < lengthSplitedDate; i++) {
+    const keyFromFormat = fromFormat[i];
+    const valueFromSplitteDate = splittedDate[i];
+
+    if (keyFromFormat === 'YYYY') {
```

Comment:
add check that it's not present in `toFormat` because you do redundant action otherwise

Reply:
ok. fix redundance with condition